### PR TITLE
Test fees in every asset

### PR DESCRIPTION
**Pull request by JBetz**
Passes an integration test that includes asset issuance, fee payment in non-native asset, and non-native asset rewards to miner. See: https://github.com/MuKnSys/gerbil-sequentia/blob/master/scripts/test-scenarios.ss#L160-L193

It's undoubtedly missing some edge cases but it should be good enough for initial testing and experimentation.

Current limitations:
- Requires using the `createrawtransaction` JSON RPC for non-native fee payments
- Fees can only be paid with a single asset

Untested:
- Mining blocks containing multiple assets
- Using block signing mechanism rather than mining

### CODE REVIEW

File: src/node/miner.cpp [hunk 1]
```diff
@@ -191,33 +192,66 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
...
-    coinbaseTx.vout.resize(1);
-    coinbaseTx.vout[0].scriptPubKey = scriptPubKeyIn;
-    coinbaseTx.vout[0].nAsset = policyAsset;
-    coinbaseTx.vout[0].nValue = nFees + GetBlockSubsidy(nHeight, chainparams.GetConsensus());
-    if (g_con_elementsmode) {
-        if(chainparams.GetConsensus().subsidy_asset != policyAsset) {
-            // Only claim the subsidy if it's the same as the policy asset.
-            coinbaseTx.vout[0].nValue = nFees;
+    if (feeMap.size() > 0) {
+        coinbaseTx.vin.resize(feeMap.size());
```
Comment (JBetz) [inline src/node/miner.cpp:196]: This should be pulled out into a separate function. It's more or less identical to the `else` block except that it iterates over the fee map to create a single output per asset.

File: src/primitives/transaction.cpp [hunk 2]
```diff
@@ -13,6 +13,7 @@
 #include <assert.h>
 
 bool g_con_elementsmode = false;
+bool g_con_sequentiamode = true;
```
Comment (JBetz) [inline src/primitives/transaction.cpp:16]: Should default to false to be consistent with elements.

File: src/primitives/transaction.h [hunk 3]
```diff
@@ -28,6 +28,9 @@ static const int SERIALIZE_TRANSACTION_NO_WITNESS = 0x40000000;
 // Globals to avoid circular dependencies.
 extern bool g_con_elementsmode;
 
+// SEQUENTIA:
+extern bool g_con_sequentiamode;
```
Comment (JBetz) [inline src/primitives/transaction.h:32]: There are two other header files where `g_con_elementsmode` is defined, so this might need to be added there as well. It would be better to only rely on one, though, and understand why elements has three.

**Review by Mixa84 — APPROVED** [review]
